Source/JavaScriptCore/runtime/CodeCache.cpp

```diff
@@ -248,7 +248,7 @@ UnlinkedFunctionExecutable* CodeCache::getUnlinkedGlobalFunctionExecutable(VM& v
     // The Function constructor only has access to global variables, so no variables will be under TDZ unless they're
```

Comment:
We should pass ImplementationVisibility to this `getUnlinkedGlobalFunctionExecutable` function. And we should include it into `SourceCodeKey`.

Reply:
Or will it be never a private visibility?

Reply:
AFAICT it seemed like all callers would expect it to be public, but I'm not super familiar with this code so I can't say for sure 😅

Reply:
looking at this again, i think sure the only caller is `constructFunction`, which i feel like should probably be public